resource/resource.go

```diff
@@ -57,6 +57,20 @@ type Extra struct {
 	Resource *unstructured.Unstructured
 }
 
+// CredentialsType is the type of credentials
```

Comment:
Nit: These comments should end with periods (I thought our linters enforced this).